File: JavaScript/Plus_One.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    len = digits.length;
    digits[len-1] += 1;
    for (var i = len-1; i >= 0; i--) {
        if (digits[i] >= 10) {
            if (i == 0) {
                digits[i] = 0;
                digits.unshift(1);
            }
            else {
                digits[i-1] += 1;
                digits[i] = 0;
            }
        }
    }
    return digits;
};

File: Python/Plus_One.py
'''
Leetcode exercise
Written by aresowj

[Plus One]
Given a non-negative number represented as an array of digits, plus one to the number.
The digits are stored such that the most significant digit is at the head of the list.
'''

class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype : List[int]
        """
        digits = digits[::-1]
        number = 0
        
        for i in range(0, len(digits)):
            number += digits[i] * (10 ** i)

        sum = number + 1
        result = []
        
        for i in str(sum):
            result.append(int(i))
            
        return result
